station-blocks-v0/src/main/java/net/modificationstation/stationapi/mixin/block/server/ServerPlayNetworkHandlerMixin.java

```diff
@@ -0,0 +1,63 @@
+package net.modificationstation.stationapi.mixin.block.server;
+
+import net.minecraft.block.Block;
+import net.minecraft.class_73;
+import net.minecraft.entity.player.ServerPlayerEntity;
+import net.minecraft.server.MinecraftServer;
+import net.minecraft.server.network.ServerPlayNetworkHandler;
+import net.minecraft.util.math.Box;
+import net.modificationstation.stationapi.api.block.HasCollisionVoxelShape;
+import net.modificationstation.stationapi.api.block.HasVoxelShape;
+import net.modificationstation.stationapi.api.util.math.Vec3i;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Redirect;
+
+@Mixin(ServerPlayNetworkHandler.class)
+public abstract class ServerPlayNetworkHandlerMixin {
+
+    @Shadow private ServerPlayerEntity field_920;
+    @Shadow private MinecraftServer field_919;
+
+    @Redirect(method = "onPlayerMove", at = @At(value = "INVOKE", target = "Lnet/minecraft/server/network/ServerPlayNetworkHandler;method_832(DDDFF)V", ordinal = 0))
```

Comment:
Redirection mixin is not ideal for compatibility. It should be replaced with a more compatible mixin if possible

Reply:
Redirect is the best way to approach this use case, because if I injected in the method it would be problematic because its called from other important places, and this method in this location is only used for correcting being inside blocks on server.